### PR TITLE
[#208] Push down limit to hbase in GSQL

### LINKED ISSUE #208: [Enhancement] Push down limit to Hbase from GSQL
**What feature are your requesting?**
Whenever limit clause is specified in GSQL query, set the hbase pageSize to push it down to hbase.

**What benefits do you foresee from the feature you are requesting?**
Optimization for limit/sample data from hbase.

**Potential solution/ideas?**
- Detect the limit clause in SQL
- Set hbase pageSize which fetches limited records from Hbase

**Additional context**

### PR DESCRIPTION
**Pull request by anishanainani**
Make sure you have checked all steps below.

### GitHub Issue
Fixes #208 

### Checklist:

- [ ] This pull request updates the documentation
- [ ] This pull request changes library dependencies
- [x] Title of the PR is of format (example) : [#25][Github] Add Pull Request Template

### What is the purpose of this pull request?
Whenever limit clause is specified in GSQL query, set the hbase pageSize to push it down to hbase

### How was this change validated?
Spark shell

```
import org.apache.spark.sql.{Column, Row, SparkSession,DataFrame}
import org.apache.spark.sql.functions._
// Create Gimel SQL reference
val gsql: (String) => DataFrame = com.paypal.gimel.sql.GimelQueryProcessor.executeBatch(_: String, spark)
gsql("set gimel.logging.level=CONSOLE")

gsql("set rest.service.method=https")
gsql("set rest.service.host=udc-service-host")
gsql("set rest.service.port=443")

// Here limit will be pushed down to HBase by setting spark.hbase.connector.pageSize
val df = gsql("select * from udc.Hbase.Test_Cluster.default.flights limit 5")
df.show
```

### Commit Guidelines
- [x] My commits all reference GH issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gimel-dataapi/gimel-sql/src/main/scala/com/paypal/gimel/sql/GimelQueryUtils.scala [hunk 1]
```diff
@@ -894,6 +894,75 @@ object GimelQueryUtils {
...
+    * For a given dataset (table) this function calls getDataSetProperties which calls catalog provider and returns the dataset properties
+    * which will be used to identify the storage of the dataset
+    *
+    * @param insertTable  - Incoming dataset
+    * @param sparkSession - spark session
+    * @param options      - Set of user options
+    * @return - Returns the storage system (hive/teradata/kafka...)
+    */
+
+  def getSystemType(insertTable: String, sparkSession: SparkSession,
```
Comment (Dee-Pac) [inline gimel-dataapi/gimel-sql/src/main/scala/com/paypal/gimel/sql/GimelQueryUtils.scala]: This should go into commons package. I believe there is already an existing function in commons.

Comment (anishanainani) [reply gimel-dataapi/gimel-sql/src/main/scala/com/paypal/gimel/sql/GimelQueryUtils.scala]: Sure, Will put it in gimel-commons

Comment (anishanainani) [reply gimel-dataapi/gimel-sql/src/main/scala/com/paypal/gimel/sql/GimelQueryUtils.scala]: The existing function is in gimel-core which this function uses and it does not make a call for getting DatasetProperties, it just resolves the system type based on storage type parameter. In GimelQueryProcessor as we are parsing SQL, it is required to first resolve the DatasetProperties for a dataset name and then resolving which type it belongs to.

**Comment by anishanainani** [comment]
@Dee-Pac I have pushed the changes. Please review it and let me know if any other change is required.

**Pull request merged**